internal/server/routes.go

```diff
@@ -205,7 +207,7 @@ func wrapRoute[Req, Res any](a *API, routeID routeIdentifier, route route[Req, R
 			return err
 		}
 
-		tx, err := a.server.db.Begin(c.Request.Context(), nil)
+		tx, err := a.server.db.Begin(c.Request.Context(), route.txnOptions)
```

Comment:
Applying the transaction options here after the authentication is checked means that a read only route can still update the user's "last seen at" time, correct?

Reply:
Exactly, any of the writes that are common to most requests (update users last seen at, update destination last seen at, and extending the access key expiry time) happen in the `authenticateRequest` function which is two calls above this. It always gets its own write transaction that it commits before returning.